index.ts

```diff
@@ -1,105 +1,40 @@
+import { EventTypes, Options, WidgetEvents, SetThemeParameters } from './types';
 
 const externalStaticOrigin = 'https://javascript.wert.io';
 
-interface options {
-  partner_id?: string
-  origin?: string
-  address?: string
-  theme?: string
-  currency?: string
-  currency_amount?: number
-  commodity?: string
-  commodity_amount?: number
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}
-
-interface extraOptions {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}
-
-interface eventOptions {
-  type: string
-  origin: string
-  data: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    [x: string]: any
-  }
-}
-
-type customListener = (data: {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}) => void;
-
-type setThemeData = {
-  theme?: string
-  colors?: {
-    [x: string]: string
-  }
-};
-
 class WertWidget {
-
   private iframe: HTMLIFrameElement = document.createElement('iframe');
-
-  partner_id?: string;
-  origin: string;
-  options: options;
-  extraOptions: extraOptions;
-  listeners: {
-    [x: string]: customListener;
-  };
-  widgetWindow: Window | null;
-  checkIntervalId: number | undefined;
-
-
-  static get eventTypes(): string[] {
-    return [
-      'close',
-      'error',
-      'loaded',
-      'payment-status',
-      'position',
-      'rate-update',
-    ];
-  }
-
-  constructor(givenOptions: options = {}) {
-    const options: options = { ...givenOptions };
-
-    if (options.container_id) {
-      console.error('container_id is no longer supported')
+  private widgetWindow: Window | null = null;
+  private checkIntervalId?: number;
+  private widget_layout_mode = 'Modal' as const
+  private await_data = false
+
+  constructor(private options: Options) {
+    // THis is required for showing error during old integration usages
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (this.options.container_id) {
+      console.error('container_id is no longer supported');
     }
 
-    this.partner_id = options.partner_id;
-    this.origin = options.origin || 'https://widget.wert.io';
-    this.extraOptions = options.extra ? { ...options.extra } : undefined;
-    this.listeners = options.listeners || {};
-    this.widgetWindow = null;
-    this.checkIntervalId = undefined;
-
-    options.widgetLayoutMode = 'Modal';
-
-    delete options.partner_id;
-    delete options.origin;
-    delete options.extra;
-    delete options.listeners;
+    if (!this.options.partner_id) {
+      throw Error("Please provide a partner_id in order for widget to work correctly");
+    }
 
-    options.await_data = (options.await_data || this.extraOptions) ? '1' : undefined;
+    if (!this.options.origin) {
+      this.options.origin = 'https://widget.wert.io';
+    }
 
-    this.options = options;
+    if (this.options.extra) {
```

Comment:
This check used to be `options.await_data || this.extraOptions `- is it safe to remove the "`options.await_data`" ?